hook/usePersistedChat.tsx

```diff
@@ -0,0 +1,138 @@
+"use client";
+
+import { useChat } from "ai/react";
+import { useState, useEffect, useCallback } from "react";
+
+export function usePersistedChat(chatId?: string) {
```

Comment:
This hook is doing too much, how can you explain it's purpose in 2 sentences? you need to simplify and deconstruct this hook, i probably doesn't to return so many attributes and callbacks, this is more complex that it should be.